packages/@glimmer/opcode-compiler/lib/opcode-builder.ts

```diff
@@ -587,9 +596,26 @@ export abstract class OpcodeBuilder<Specifier> {
     return params.length;
   }
 
-  compileArgs(params: Option<WireFormat.Core.Params>, hash: Option<WireFormat.Core.Hash>, synthetic: boolean) {
+  compileArgs(params: Option<WireFormat.Core.Params>, hash: Option<WireFormat.Core.Hash>, blocks: Option<Blocks>, synthetic: boolean) {
+    if (blocks) {
+      this.pushYieldableBlock(blocks.main);
+      this.pushYieldableBlock(blocks.else);
+      this.pushYieldableBlock(blocks.attrs);
+    }
+
     let count = this.compileParams(params);
 
+    let flags = count << 4;
```

Comment:
I'm a bit worried that the encoding we are introducing is spread between the builder and the encoder. I'm wondering if we should move all the encoding to the encoder so we have a better idea of heap layout and the available and reserved size of different types of operands.